build/org.eclipse.cdt.managedbuilder.core.tests/tests/org/eclipse/cdt/managedbuilder/core/tests/CompilationDatabaseGenerationTest.java

```diff
@@ -0,0 +1,189 @@
+	}
+
+	/**
+	 * Tests format for compile_commands.json. JSON array is expected, containing an element for the c file
+	 * @throws JsonIOException
+	 * @throws CoreException
+	 */
+	@Test
+	public void testJsonFormat() throws JsonIOException, CoreException {
+		setWorkspace("regressions");
+		final IProject app = loadProject("helloworldC");
+		isGenerateFileOptionEnabled(true);
+		app.build(IncrementalProjectBuilder.FULL_BUILD, null);
+		IFile commandsFile = app.getFile("build/compile_commands.json");
+		if (commandsFile.exists()) {
+
+			try (FileReader reader = new FileReader(commandsFile.getLocation().toFile())) {
+				Gson gson = new Gson();
+				JsonArray jsonArray = gson.fromJson(reader, JsonArray.class);
+				System.out.println(jsonArray);
+				for (JsonElement element : jsonArray) {
+					CompilationDatabaseInformation compileCommand = gson.fromJson(element,
+							CompilationDatabaseInformation.class);
+
+					assertTrue(compileCommand.directory() != null && !compileCommand.directory().isEmpty());
+					assertTrue(compileCommand.command() != null && !compileCommand.command().isEmpty());
+					assertTrue(compileCommand.file() != null && !compileCommand.file().isEmpty());
+					assertTrue(compileCommand.file().endsWith("src/helloworldC.c"));
+				}
+
+			} catch (IOException e) {
+				assertTrue(false);
+			}
+
+		}
+	}
+
+	/**
+	 * Test that compile_commands.json is correctly generated when more than one .c file is present as a source file
+	 * @throws CoreException
+	 */
+	@Test
+	public void testMultipleFiles() throws CoreException {
+		setWorkspace("regressions");
+		final IProject app = loadProject("helloworldC");
+		IFile aFile = ManagedBuildTestHelper.createFile(app, "src/newFile.c");
+		isGenerateFileOptionEnabled(true);
+		app.build(IncrementalProjectBuilder.FULL_BUILD, null);
+		IFile commandsFile = app.getFile("build/compile_commands.json");
+		int numberOfElementsFound = 0;
+		boolean helloworldCIsPresent = false;
+		boolean newFileIsPresent = false;
+		try (FileReader reader = new FileReader(commandsFile.getLocation().toFile())) {
+			Gson gson = new Gson();
+			JsonArray jsonArray = gson.fromJson(reader, JsonArray.class);
+			System.out.println(jsonArray);
+			for (JsonElement element : jsonArray) {
+				CompilationDatabaseInformation compileCommand = gson.fromJson(element,
+						CompilationDatabaseInformation.class);
+				numberOfElementsFound++;
+				if (compileCommand.file().endsWith("helloworldC.c")) {
+					helloworldCIsPresent = true;
+				}
+				if (compileCommand.file().endsWith("newFile.c")) {
+					newFileIsPresent = true;
+				}
+			}
+			assertEquals(2, numberOfElementsFound);
+			assertTrue(helloworldCIsPresent);
+			assertTrue(newFileIsPresent);
+		} catch (IOException e) {
+			assertTrue(false);
+		}
+
+	}
+
+	/**
+	 * Tests that cpp files are handled by compile_commands.json file generator
+	 * @throws CoreException
+	 */
+	@Test
+	@Ignore("This will be temporary skipped due to builder error")
```

Comment:
The difference between what you see in the workspace and what happens on the build machine is probably down to different sets of enabled bundles. See https://github.com/eclipse-cdt/cdt/blob/main/TESTING.md#using-limited-sets-of-plug-ins-in-tests (point 3)